disjoint_set/main.py

```diff
@@ -44,6 +44,13 @@ def itersets(self) -> Generator[Set[T], None, None]:
         for element_class in element_classes.values():
             yield element_class
 
+    def _legacy_find(self, x: T) -> T:
```

Comment:
I'm not sure there's much point in leaving this logic here - I agree that non-recursive approach is better and from the functional point of view the find() method will stay the same, so we can just refactor the existing method.